src/protocol/protobuf_client.py

```diff
@@ -604,11 +604,17 @@ async def _process_user_stats_response(self, body):
             achi_block_enum = 32 * (achievement_block.achievement_id - 1)
             for index, unlock_time in enumerate(achievement_block.unlock_time):
                 if unlock_time > 0:
-                    if str(achievement_block.achievement_id) not in achievements_schema[str(game_id)]['stats'] or \
-                            str(index) not in achievements_schema[str(game_id)]['stats'][str(achievement_block.achievement_id)]['bits']:
-                        logger.warning("Non existent achievement unlocked")
-                        continue
                     try:
+                        if 'bits' not in achievements_schema[str(game_id)]['stats'][str(achievement_block.achievement_id)]:
+                            logger.warning("'bits' not found in achievements_schema while parsing achievement %d for game %s", index, game_id)
+                            logger.info(achievements_schema)
+                            continue
```

Comment:
Isn't it also the case of "Non existent achievement unlocked" that is copied below?

Have you identified if all real, unlocked achievements for that game are handled?

Reply:
Unfortunately, the "Non existent achievement unlocked" test below does not work, since it uses the 'bits' part in its condition, while it may not even be present. Before my patch, this test is also executed outside the 'try' block, which causes the achievement import to fail entirely for all games.

Maybe this is valuable to know: I earned train simulator achievements years ago, before they decided to replace them all with new achievements. There might be a mismatch somewhere because of this.
 
I asked a friend who owns train simulator to verify that his new achievements work, I will keep you up to date.

Reply:
> Unfortunately, the "Non existent achievement unlocked" test below does not work, since it uses the 'bits' part in its condition, while it may not even be present. Before my patch, this test is also executed outside the 'try' block, which causes the achievement import to fail entirely for all games.

yes, I understand why it fails, I just wonder it if it is _logically_ the case of "Non existent achievement unlocked".

> Maybe this is valuable to know: I earned train simulator achievements years ago, before they decided to replace them all with new achievements. There might be a mismatch somewhere because of this.

Yeah, good to know, if this is the reason then that is more less the case of "non existent (in schema) achievement unlocked". So you have unlocked achievement that is no longer valid/used achievement in the game.

I'm merging your change and then I'll refactor a little bit the this the previous code. BTW parsing should be placed definitely outside `protobuf_client`.

Thanks!

Reply:
I guess it indeed logically means "Non existent achievement unlocked".

My friend reported back on train simulator and the existing achievements work. Here is the proof: [before+after.zip](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/files/6290366/before%2Bafter.zip)
- **Before** was designed to crash the plugin and removes all achievements (if you remove the steam plugin and then use this version to add it again)
- **After** is the code I requested a pull for, and fixes the achievements